colibre/registration.py

```diff
@@ -24,333 +24,458 @@
+            ),
+        )
 
-# Get HI to dust ratio
-try:
-    dust_to_hi_ratio = total_dust_mass / HI_mass
-    dust_to_hi_ratio.name = "M_{{\\rm dust}}/M_{{\\rm HI}}"
-    setattr(self, "gas_dust_to_hi_ratio", dust_to_hi_ratio)
-
-except AttributeError:
-    # We did not produce these quantities.
-    setattr(
-        self,
-        "gas_dust_to_hi_ratio",
-        unyt.unyt_array(
-            unyt.unyt_array(np.ones(np.size(catalogue.masses.m_gas)), "dimensionless"),
-            name="$M_{{\\rm dust}}/M_{{\\rm HI}}$ not found, default to 1s",
-        ),
-    )
+    return
 
 
-# Get H2 masses
-try:
-    gas_mass = catalogue.masses.m_gas
-    H_frac = getattr(catalogue.element_mass_fractions, "element_0")
+def register_h2_masses(self, catalogue):
 
-    # Try CHIMES arrays
-    if hasattr(catalogue.species_fractions, "species_7"):
-        H2_frac = getattr(catalogue.species_fractions, "species_7")
-    # If species_7 doesn't exist, switch to the (default) Table-cooling case
-    else:
-        H2_frac = getattr(catalogue.species_fractions, "species_2")
-
-    H2_mass = gas_mass * H_frac * H2_frac * 2.0
-    H2_mass_wHe = gas_mass * nonmetal_frac * H2_frac * 2.0
-    H2_mass.name = "$M_{\\rm H_2}$"
-
-    setattr(self, "gas_H2_mass", H2_mass)
-    setattr(self, "gas_H2_plus_He_mass", H2_mass_wHe)
-
-except AttributeError:
-    # We did not produce these quantities.
-    setattr(
-        self,
-        "gas_H2_mass",
-        unyt.unyt_array(
-            catalogue.masses.m_gas, name="$M{\\rm H_2}$ not found, showing $M_{\\rm g}$"
-        ),
-    )
+    # Define aperture size in kpc
+    aperture_sizes = [30, 100]
 
-# Get neutral H masses and fractions
-try:
-    gas_mass = catalogue.masses.m_gas
-    H_frac = getattr(catalogue.element_mass_fractions, "element_0")
+    for aperture_size in aperture_sizes:
 
-    # Try CHIMES arrays
-    if hasattr(catalogue.species_fractions, "species_7"):
-        HI_frac = getattr(catalogue.species_fractions, "species_1")
-        H2_frac = getattr(catalogue.species_fractions, "species_7")
-    # If species_7 doesn't exist, switch to the (default) Table-cooling case
-    else:
-        HI_frac = getattr(catalogue.species_fractions, "species_0")
-        H2_frac = getattr(catalogue.species_fractions, "species_2")
+        # Fetch H2 mass
+        H2_mass = getattr(catalogue.gas_species_masses, f"H2_mass_{aperture_size}_kpc")
+
+        # Label of the derived field
+        H2_mass.name = f"$M_{{\\rm H_2}}$ ({aperture_size} kpc)"
 
-    HI_mass = gas_mass * H_frac * HI_frac
-    H2_mass = gas_mass * H_frac * H2_frac * 2
-    neutral_H_mass = HI_mass + H2_mass
-    neutral_H_mass.name = "$M_{\\rm HI + H_2}$"
+        # Register field
+        setattr(self, f"gas_H2_mass_{aperture_size}_kpc", H2_mass)
 
-    setattr(self, "gas_neutral_H_mass", neutral_H_mass)
+    return
 
+
+def register_cold_gas_mass_ratios(self, catalogue):
+
+    # Define aperture size in kpc
+    aperture_sizes = [30, 100]
```

Comment:
Can we make this a parameter to the function, seeing as we use the same apertures everywhere?